### PR TITLE
Remove TSample

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/dataset.py [hunk 1]
```diff
@@ -1240,7 +1240,7 @@ def export_genotypes(self, output, expr, types=False, export_ref=False, export_m
 
         Export the same information without identifiers, resulting in a file with no header:
 
-        >>> vds.export_genotypes('output/genotypes.tsv', 's, v, s.id, g.dp, va.anno1, va.anno2')
+        >>> vds.export_genotypes('output/genotypes.tsv', 's, v, s, g.dp, va.anno1, va.anno2')
```
Comment (jbloom22) [inline python/hail/dataset.py]: Should be `s, v, g.gq, g.dp, va.anno1, va.anno2`

File: python/hail/docs/tutorial.ipynb [hunk 2]
```diff
@@ -62,7 +62,7 @@
    },
    "outputs": [],
    "source": [
-    "from hail import *\n",
+    "from hail import *      \n",
```
Comment (jbloom22) [inline python/hail/docs/tutorial.ipynb]: stray tab?

File: src/main/scala/is/hail/expr/Parser.scala [hunk 3]
```diff
@@ -533,7 +533,7 @@ object Parser extends JavaTokenParsers {
       "Float" ^^ { _ => TFloat } |
       "Double" ^^ { _ => TDouble } |
       "String" ^^ { _ => TString } |
-      "Sample" ^^ { _ => TSample } |
+      "Sample" ^^ { _ => TString } | // FIXME back-compatibility
```
Comment (jbloom22) [inline src/main/scala/is/hail/expr/Parser.scala]: When will this be cut?

**Comment by jbloom22** [comment]
Great change, just a couple typos you can fix while rebasing.

File: src/main/scala/is/hail/expr/Parser.scala (hunk 3 above)
Comment (tpoterba) [inline src/main/scala/is/hail/expr/Parser.scala]: the next file format increment, I guess. If somebody somewhere has a VDS with a sample as a variant or sample annotation, this line is required to maintain compatibility

File: python/hail/docs/tutorial.ipynb (hunk 2 above)
Comment (tpoterba) [inline python/hail/docs/tutorial.ipynb]: I absolutely hate intellij's ipynb interface

File: python/hail/dataset.py (hunk 1 above)
Comment (tpoterba) [inline python/hail/dataset.py]: good catch, I wasn't even looking at the things I changed. Will fix

**Comment by tpoterba** [comment]
back to you @jbloom22

**Comment by tpoterba** [comment]
Need this to make another branch pass / easier to use

**Pull request merged**